pkg/virt-handler/realtime.go

```diff
@@ -194,3 +189,22 @@ func schedSetScheduler(pid int, policy policy, param schedParam) error {
 	}
 	return nil
 }
+
+func (c cpuMask) isRealtimeVCPU(vcpuID string) bool {
+	if len(c.mask) == 0 {
+		return true
+	}
+	if t, ok := c.mask[vcpuID]; ok {
+		return t == enabled
+	}
+	return false
+}
+
+func (c *cpuMask) has(vcpuID string) bool {
+	_, ok := c.mask[vcpuID]
+	return ok
+}
```

Comment:
Note that this function only checks the key's existence, but doesn't look at the value at all. Is this intentional?

For example, if my mask is `^6,4-8` - is `6` expected to be enabled?
Or more generalized - what has precedence - the first appearance or the last?

Reply:
Yes, it's intentional. The reason is that it needs to check only when attempting to define an index as enabled to ensure that it is not overwriting the previous disabled value. When disabling, we don't check if it has already been defined since it is the only case when we overwrite whatever is stored.

So in that sequence you wrote, first it will disable the 6th core and then, when trying to enable it for the range, it will check if it has already been defined and skip it in this case.
The outcome would be the same even if you wrote `4-8,^6`.